src/tables/svg.js

```diff
@@ -0,0 +1,110 @@
+import { Parser } from '../parse.js';
+import table from '../table.js';
+
+// https://learn.microsoft.com/en-us/typography/opentype/spec/svg
+
+/**
+ * @typedef {Map<number, Uint8Array>} SVGTable
+ */
+
+/**
+ * @param {DataView} data
+ * @param {number} offset
+ * @returns {SVGTable}
+ */
+function parseSvgTable(data, offset) {
+    const svgTable = new Map();
+    const buf = data.buffer;
+    const p = new Parser(data, offset);
+    const version = p.parseUShort();
+    if (version !== 0) return svgTable;
+
+    p.relativeOffset = p.parseOffset32();
+    const svgDocumentListOffset = data.byteOffset + offset + p.relativeOffset;
+    const numEntries = p.parseUShort();
+    const svgDocMap = new Map();
+    for (let i = 0; i < numEntries; i++) {
+        const startGlyphID = p.parseUShort();
+        const endGlyphID = p.parseUShort();
+        const svgDocOffset = svgDocumentListOffset + p.parseOffset32();
+        const svgDocLength = p.parseULong();
+        let svgDoc = svgDocMap.get(svgDocOffset);
+        if (svgDoc === undefined) {
```

Comment:
I think "cutting" the svg document out of the buffer would be an excellent candidate for deferred parsing as discussed here: 
https://github.com/opentypejs/opentype.js/pull/698#issuecomment-2046612897
https://github.com/opentypejs/opentype.js/pull/698#issuecomment-2046626397

See here for the code for gvar parsing:
https://github.com/opentypejs/opentype.js/blob/be0d4417a04d92d43178e075273048e926164abf/src/parse.js#L873

Because with many and more complex SVG documents, right now everything would be parsed into memory at once, even though it might not be needed at all, bloating the memory unnecessarily.

Reply:
Just to be sure we are on the same page. By "cutting the svg document out of the buffer" you  mean `svgDoc = new Uint8Array(buf, svgDocOffset, svgDocLength);`? As I understand this is a cheap operation, because no memory is copied. So we spend only 24 bytes per document here (`sizeOfPointer(buf) + sizeOfNumber(svgDocOffset) + sizeOfNumber(svgDocLength)`). An SVG table consumes `(sizeOfNumber(glyphId) + sizeOfPointer(svgDoc)) * countOfSVGGlyphs + 24 * countOfSVGDocuments`. Therefore, an SVG table with 1000 SVG documents with one glyph per document would be about 40kb. Is it worth to parse it lazily?
Or did you mean something different by "an excellent candidate for deferred parsing"?

Reply:
I'm not too familiar with the internals of js regarding data views and typed arrays, but if the array is really just a pointer to that part of the DataView in memory, then of course that's totally fine!

Reply:
Yes, when constructed with an ArrayBuffer as the first parameter, an Uint8Array points to the specified buffer. Here is the relevant doc https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#parameters

Reply:
I learned something today, thanks! :)